### PR TITLE
[IoTCentral] `az iot central app`: Address several S360 fixes

### PR DESCRIPTION
**Pull request by poanchen**
**Description**

The main purpose of this PR is to address several S360 fixes with the inconsistency between our swagger and our RP response. This PR will also make sure our command az iot central app will be using the latest python packages that we just [released](https://pypi.org/project/azure-mgmt-iotcentral/4.1.0). The S360 issue included resourceType should be camel case, appTemplateName field should be name in appTemplate API, add new industry and locations field in appTemplate API, add new origin and properties field in operations API, remove the need of checking etag when updating the existing iotc app command.

Note: By removing the etag check in az cli iot central app update command, it allows us to remove the etag field from RP response later on. Otherwise the az cli iot central app update command will effectively fail.

**Testing Guide**

We have re-record the tests with live validation and it passed.

**History Notes**

[IoTCentral] `az iot central app`: Address several S360 fixes
[IoTCentral] `az iot central app update`: Remove the need of checking etag when updating the existing iotc app.
[IoTCentral] Change the resourceType (IotApps) to be in camel case.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
IoTCentral

**Comment by poanchen** [comment]
@zhoxing-ms please let us know what we need to do to complete this so that we can make it to the S183 milestone.

File: src/azure-cli/HISTORY.rst
Comment (digimaun) [inline src/azure-cli/HISTORY.rst:6]: It appears most of these fixes/changes are a function of updating the SDK. Is that right?

**Review by digimaun — APPROVED** [review]

File: src/azure-cli/HISTORY.rst
Comment (poanchen) [reply src/azure-cli/HISTORY.rst:6]: Yes, not much changes we need to do on az cli side other than updating the SDK except the etag.

**Comment by poanchen** [comment]
will we be able to make it to the S183 milestone with Code complete: 02/24/2021? @digimaun @yonzhan

**Comment by digimaun** [comment]
Once @zhoxing-ms approves we should be able to merge.

**Comment by yonzhan** [comment]
@zhoxing-ms please help with this PR with high priority.

File: src/azure-cli/HISTORY.rst
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:8]:
```suggestion
* Ensure the resourceType (IotApps) is in camel case.
```

**Comment by poanchen** [comment]
ok. Updated the HISTORY as suggested.

File: src/azure-cli/HISTORY.rst
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:7]: Because our current pipeline automatically generate history notes from PR tile and descriptions, so please don't write the history into history note manually, it should be written into the PR description instead.
If you want more details, please refer to the doc: [link](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests)

File: src/azure-cli/HISTORY.rst
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:13]: Because our current pipeline automatically generate history notes from PR tile and descriptions, so please don't write the history into history note manually, it should be written into the PR description instead.
If you want more details, please refer to the doc: [link](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests)

File: src/azure-cli/azure/cli/command_modules/iot/custom.py
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/iot/custom.py:1094]: May I ask why we don't need `etag` anymore?

Comment (poanchen) [reply src/azure-cli/azure/cli/command_modules/iot/custom.py:1094]: We got S360 issue warning us that swagger doesn't match with our RP response. We then removed the etag from our RP. After through testing, we realized that it will potentially break our az cli iot central app update command if the changes get out to prod. In order for RP changes to get out, we need az cli changes (without checking the etag) out first so that az cli iot central update command will continue to work. That is why we remove it now.

Comment (poanchen) [reply src/azure-cli/azure/cli/command_modules/iot/custom.py:1094]: we also have etag in our header just in case we needed it.

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/iot/custom.py:1094]: Got it, thanks~

File: src/azure-cli/HISTORY.rst
Comment (poanchen) [reply src/azure-cli/HISTORY.rst:13]: will remove

**Review by zhoxing-ms — APPROVED** [review]

**Comment by poanchen** [comment]
Looks like we got all our approvals. Please merge this PR whenever you guys feel comfortable. Thanks!

**Pull request merged**
